### PR TITLE
feat/auth: implement basic auth types

### PR DESCRIPTION
**Pull request by vinipsmaker**

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @ustulation

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/auth/mod.rs [hunk 1]
```diff
@@ -0,0 +1,91 @@
+
+use routing::XorName;
```
Comment (ustulation) [inline src/auth/mod.rs]: Copyright should be the first thing.

File: src/auth/ffi.rs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    pub id: *mut u8,
+    /// TODO: doc
+    pub id_len: usize,
+    /// TODO: doc
+    pub id_cap: usize,
+
+    /// TODO: doc
+    pub scope: *mut u8, // null if not present
+    /// TODO: doc
+    pub scope_len: usize, // 0 if above is null
```
Comment (ustulation) [inline src/auth/ffi.rs]: Move the inline comment to the Doc comment (Keep the TODO)

File: src/auth/ffi.rs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    /// TODO: doc
+    pub scope_cap: usize, // 0 if above is null
```
Comment (ustulation) [inline src/auth/ffi.rs]: Move the inline comment to the Doc comment (Keep the TODO)

File: src/auth/mod.rs (hunk 1 above)
Comment (vinipsmaker) [inline src/auth/mod.rs]: `rustfmt` got confused because accidentally I've put an additional `/` on the first copyright line. Thanks. Fixing...

**Pull request merged**